### DISCUSSION TITLE
Option to add num_threads options to PstFrom.add_parameters?

### DISCUSSION THREAD

**Discussion by kbefus** [pypest/pyemu#537] [Ideas]
I'm having a sporadic issue where I get hung up on "starting interp point loop for x points", and it looks like the unstructured mesh I'm using is defaulting to num_threads=10 and the utils.geostats.OrdinaryKrige._calc_factors_mp call instead of the single thread org option. It looks like my unstructured (DISV) grid is catching a True here:

https://github.com/pypest/pyemu/blob/b301698b6d2e0491fa6e0f432ff851464c8b2d5f/pyemu/utils/pst_from.py#L2652

and then num_threads=10 is getting stuck in my system. In my local copy of pyemu, I put in a 1 here (and added verbose=True) and then the hang up went away using the org routine instead of the mp one. I don't think I can replicate this with a simple example, but seems like a multiprocessing problem on my machine. Would be great to have a num_threads kwarg or similar that could be passed through add_parameters that would trickle to the underlying calls.

**Comment by jtwhite79** [comment]
Yeah it's a problem.  We've just about got [pypestutils](https://github.com/pypest/pypestutils) baked in so that if it's available, it will be used to solve the kriging equations (which is heaps faster bc it's compiled fortran).  But we probably should sneak in a num_threads option too...

**Reply by christianlangevin** [reply]
+1 for Fortran!

**Reply by kbefus** [reply]
Sweet.. Thanks :). It's also me breaking things on my first time through with my own model after watching heaps of your videos and goofing around with notebooks. I haven't gone down the route of figuring out why my disv grid is switching the structured flag to true either.
